pkg/operator/ceph/csi/version.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package csi
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// CephCSIVersion represents the Ceph CSI version format
+type CephCSIVersion struct {
+	Major  int
+	Minor  int
+	Bugfix int
+}
+
+var (
+	//minimum supported version is 2.0.0
+	minimum = CephCSIVersion{2, 0, 0}
+
+	//supportedCSIVersions are versions that rook supports
+	supportedCSIVersions = []CephCSIVersion{minimum}
+
+	// for parsing the output of `cephcsi`
+	versionCSIPattern = regexp.MustCompile(`v(\d+)\.(\d+)\.(\d+)`)
+)
+
+func (v *CephCSIVersion) String() string {
+	return fmt.Sprintf("%d.%d.%d",
+		v.Major, v.Minor, v.Bugfix)
+}
+
+// Supported checks if the detected version is part of the known supported CSI versions
+func (v *CephCSIVersion) Supported() bool {
+	if v.isAtLeast(&minimum) == false {
+		return false
+	}
+
+	// if AllowUnsupported is set also csi-images greater than the supported ones are allowed
+	if AllowUnsupported {
+		return true
+	}
+	for _, sv := range supportedCSIVersions {
+		if v.Major == sv.Major {
+			if v.Minor == sv.Minor {
+				if v.Bugfix >= sv.Bugfix {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+func (v *CephCSIVersion) isAtLeast(version *CephCSIVersion) bool {
+	if v.Major > version.Major {
```

Comment:
to avoid multiple if blocks and complexity can we write this as
```go
if v.Major > version.Major {
	return true
}
if v.Major == version.Major && v.Minor >= version.Minor {
  if v.Minor > version.Minor{
      return true
   }
   if v.Bugfix >= version.Bugfix {
	return true
   }
}

return false
```

Reply:
I do not care but I had the impression, according to e.g. [`isSuperior`](https://github.com/rook/rook/blob/master/pkg/operator/ceph/version/version.go#L208), this is some kind of a style guide?

Reply:
@travisn can this be changed as suggested?

Reply:
Changed it the way @Madhu-1 suggested it.